Lista2/4.c

```diff
@@ -63,70 +53,116 @@ bool isOperator(char el){
     case '/':
         return true;
         break;
-    
+
     default:
         return false;
         break;
     }
-
 }
 
-int applyOperator(char operator, char n1, char n2){
+int applyOperator(char operator, int n1, int n2)
+{
     switch (operator)
     {
     case '+':
-        return (int)n1 + (int)n2;
+        return n1 + n2;
         break;
 
     case '-':
-        return (int)n1 - (int)n2;
+        return n1 - n2;
         break;
 
     case '*':
-        return (int)n1 * (int)n2;
+        return n1 * n2;
         break;
 
     case '/':
-        return (int)n1 / (int)n2;
+        return n1 / n2;
         break;
-    
+
     default:
         return 0;
         break;
     }
 }
 
-int searchingResult(stQueue *queue){
-    int twoInRow = 0, result = 0;
-    stQueue *functionQueue;
 
-    for(queue->start; queue->start < queue->end ; queue->start++){
+void showQueue(stQueue *queue)
+{
+    for (int i = 0; i < queue->end; i++)
+    {
+        if(isOperator(queue->array[i])){
+            printf("|%c", queue->array[i]);
+        }else{
+        printf("|%d", queue->array[i]);
+        }
+    }
 
+    printf("\n");
+}
 
-        if(isOperator(queue->start)){//se for constatado que é um operador
+
+void searchingResult(stQueue *queue)
+{
+    int twoInRow = 0;
+    void *dequedOperator = NULL , *dequedNumber1= NULL , *dequedNumber2 = NULL;
+    bool haveOperator = false;
+
+   for(queue->start; queue->start <= queue->end && queue->end < MAX; queue->start){
+    showQueue(queue);
+    if(isOperator(queue->array[queue->start])){//quando sabemos que é operador
+     if(haveOperator){//se já tinha um operador antes, ele é inserido ao final
+        enqueue(queue, dequedOperator);
```

Comment:
Aqui voce esta mandando NULL para a fila?

Reply:
não não, nesse caso eu mando um operador que não será usado no momento para o fim da fila. Seria um caso assim: ` + -`

Como o menos é sucessor direto do +, ele só vai ser usado depois que o - for aplicado, então jogo o + para o fim da fila

Reply:
Entendi. É pq voce inicia essa variavel "dequedOperator" como NULL, só que não faz uma verificação nela ou insere algum endereço, ai pensei que estava mandando como NULL.

Reply:
Na verdade é como se tivessem duas verificações que pegam isso, tem a função isOperator, que é a primeira que eu chamo, se for passado NULL essa função retorna false e vai ser pra saber que não é operador, então não vai entrar aí em nenhum desses casos. Além disso, dá pra interpretar a booleana haveOperator como verificação pra isso também, se ela for false eu sei que não tem operador nenhum e não vai entrar no enqueue